### PR TITLE
HDDS-10626. [LeaseRecovery] OM shuts down with 'SecretKey client must have been initialized already'

### PR DESCRIPTION
**Pull request by ChenSammi**
## What changes were proposed in this pull request?

OzoneManager crashed during reapply Ratis transactions during startup.  During recoverLease reapply, the request wants to generate the block token which depends on SecretKey service, while SecretKey client is not instantiated at that time because OzoneManager is reapply txs, and all the other services are not yet initialized.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10626

## How was this patch tested?

Tested in the cluster where the issue is found and can repeatedly reproduced.

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/file/TestOMRecoverLeaseRequest.java [hunk 1]
```diff
@@ -356,6 +359,35 @@ public void testLeaseSoftLimitForHsyncRecoverFile(boolean force) throws Exceptio
...
+
+    // set OM as Not Running and enable block token
+    when(ozoneManager.isRunning()).thenReturn(false);
+    when(ozoneManager.isGrpcBlockTokenEnabled()).thenReturn(true);
+
+    GenericTestUtils.LogCapturer logCapturer = GenericTestUtils.LogCapturer.captureLogs(OMRecoverLeaseRequest.LOG);
+
+    OMClientResponse omClientResponse = validateAndUpdateCache();
+    OMResponse omResponse = omClientResponse.getOMResponse();
+    assertEquals(OzoneManagerProtocolProtos.Status.OK, omResponse.getStatus());
```
Comment (jojochuang) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/file/TestOMRecoverLeaseRequest.java:377]: is it okay to just let it return OK? Or is it better to return an error? Ideally the client should keep retrying until OM is running.

Comment (ChenSammi) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/file/TestOMRecoverLeaseRequest.java:377]: This pass is only for ratis raft log reapply, so skip the token is OK.  But I find a more clean way to handle this. I will upload a new patch.

Comment (ChenSammi) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/file/TestOMRecoverLeaseRequest.java]: The new patch is tested by Pratyush, and it works, the issue doesn't happen again(It's reproducible) .

**Review by ashishkumar50 — COMMENTED** [review]
Thanks @ChenSammi for the patch, Change LGTM +1.

**Review by jojochuang — APPROVED** [review]
merging it

**Comment by ChenSammi** [comment]
Thanks @jojochuang and @ashishkumar50 for the review.

**Pull request merged**